mm.c

```diff
@@ -623,6 +648,40 @@ static void test_place_block_into_free_list()
     printf("Test passed.\n\n");
 }
 
+static void test_extend_heap() {
+	printf("Test extend_heap.\n");
+
+	char **test_heap_ptr;
+	char **initial_epilog_location;
+	char **initial_heap_end;
+	char **new_epilog_location;
+	size_t initial_heap_size = 8; // words
+	size_t new_block_size = 50; // words
+
+	// create initial heap and epilog taken space at the end
+	test_heap_ptr = mem_sbrk((initial_heap_size) * WORD_SIZE);
```

Comment:
I guess what can be done is not extend the heap but use the heap that was created from `init`. Unfortunately, we need `extend_heap` to make `init`. It is kind of a circular dependency.

Since we are working with the total heap of the process and testing that `extend_heap` managed the new heap space well, I am not sure how to not screw up the execution of the whole program, as even calling `extend_heap` might interfere with the lab's tests.

Overall, we will need to remove all `malloc` from our code in the end anyway so all tests will go out of the lab. In conjunction with that spirit we can keep this test only while we are working on the helper functions and then disable it when making the final functions.